### ISSUE TITLE
model.predict_generator & model.fit [Keras] don't give the same prediction for the same image !!

### ISSUE THREAD

**Issue by Delarti** [tensorflow/tensorflow#41207]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
N/A (google colab)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
n/a
- TensorFlow installed from (source or binary):
binary
- TensorFlow version (use command below):
1.15.2
- Python version:
3.6.9
- Keras version:
2.1.6

**Describe the current behavior**

Hi everyone,

I trained a model using Keras.model.fit_generator. However, when I use model.predict_generator(...) and model.predict(...), everything works well BUT I don't get the same predictions for the same image... I am aware of a similar problem [https://github.com/keras-team/keras/issues/3477](url) but I tried everything that was suggested (no augmentation used, shuffle=False, renamed my files,...) and still get different results... Anyone could help please?

**Describe the expected behavior**
I expect to have the same results...

**Comment by Saduf2019** [comment]
@Delarti 
Please share simple stand alone indented code for us to replicate the issue faced or if possible share colab gist with error faced.
